{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import demultiplexed filtered reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Data/Demux/SeqRun_2’: File exists\n",
      "mkdir: cannot create directory ‘Data/Demux/SeqRun_2’: File exists\n",
      "mkdir: cannot create directory ‘Data/Demux/SeqRun_2’: File exists\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "met = pd.read_csv('metadata.tsv', sep='\\t', index_col='#SampleID')\n",
    "\n",
    "# sort reads by sequencing run\n",
    "for ri in set(met.Run_index):\n",
    "  tmp = met.loc[met.Run_index == ri]['SeqRun'].tolist()[0]\n",
    "  !mkdir Data/Demux/$tmp\n",
    "  !mv Data/Demux/$ri/* Data/Demux/$tmp\n",
    "  !rm -r Data/Demux/$ri"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mImported Data/Demux/SeqRun_2 as CasavaOneEightSingleLanePerSampleDirFmt to Data/Demux/SeqRun_2_demux.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "met = pd.read_csv('metadata.tsv', sep='\\t', index_col='#SampleID')\n",
    "\n",
    "# import reads\n",
    "for sr in set(met.SeqRun):\n",
    "  demux = 'Data/Demux/%s_demux.qza'%sr\n",
    "  \n",
    "  !qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "    --input-path Data/Demux/$sr \\\n",
    "    --input-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "    --output-path $demux"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Data/Demux/SeqRun_2_demux.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "for sr in set(met.SeqRun):\n",
    "  demux = 'Data/Demux/%s_demux.qza'%sr\n",
    "  deqzv = 'Data/Demux/%s_demux.qzv'%sr\n",
    "  \n",
    "  !qiime demux summarize \\\n",
    "    --i-data $demux \\\n",
    "    --o-visualization $deqzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved SampleData[PairedEndSequencesWithQuality] to: Data/Cutadapt/SeqRun_2_trim.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Cutadapt/SeqRun_2_trim.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "met = pd.read_csv('metadata.tsv', sep='\\t', index_col='#SampleID')\n",
    "adapter = met.LinkerPrimerSequence.tolist()[0]\n",
    "\n",
    "!mkdir Data/Cutadapt\n",
    "for sr in set(met.SeqRun):\n",
    "  demux = 'Data/Demux/%s_demux.qza'%sr\n",
    "  cutad = 'Data/Cutadapt/%s_trim.qza'%sr\n",
    "  cuqzv = 'Data/Cutadapt/%s_trim.qzv'%sr\n",
    "  \n",
    "  !qiime cutadapt trim-paired \\\n",
    "    --i-demultiplexed-sequences $demux \\\n",
    "    --o-trimmed-sequences $cutad \\\n",
    "    --p-cores 10 \\\n",
    "    --p-anywhere-f $adapter \\\n",
    "    --p-error-rate 0.2 \\\n",
    "    --p-match-adapter-wildcards \\\n",
    "    --p-discard-untrimmed \\\n",
    "    --p-match-read-wildcards\n",
    "\n",
    "  # Visualization\n",
    "  !qiime demux summarize \\\n",
    "      --i-data $cutad \\\n",
    "      --o-visualization $cuqzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved SampleData[JoinedSequencesWithQuality] to: Data/Joined/SeqRun_2_joined.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#import pandas as pd\n",
    "#met = pd.read_csv('metadata.tsv', sep='\\t', index_col='#SampleID')\n",
    "#\n",
    "#!mkdir Data/Joined\n",
    "#for sr in set(met.SeqRun):\n",
    "#  cutad = 'Data/Cutadapt/%s_trim.qza'%sr\n",
    "#  joined = 'Data/Joined/%s_joined.qza'%sr\n",
    "#  !qiime vsearch join-pairs \\\n",
    "#    --i-demultiplexed-seqs $cutad \\\n",
    "#    --p-threads 8 \\\n",
    "#    --p-maxdiffs 12 \\\n",
    "#    --o-joined-sequences $joined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Deblur/SeqRun_2_bac_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: Data/Deblur/SeqRun_2_bac_rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved DeblurStats to: Data/Deblur/SeqRun_2_bac_denoising-stats.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Deblur/SeqRun_2_bac_rep-seqs.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Deblur/SeqRun_2_bac_table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#import pandas as pd\n",
    "#met = pd.read_csv('metadata.tsv', sep='\\t', index_col='#SampleID')\n",
    "#\n",
    "#adapter = len(met.LinkerPrimerSequence.tolist()[0])\n",
    "#!mkdir Data/Deblur\n",
    "#for sr in set(met.SeqRun):\n",
    "#  joined = 'Data/Joined/%s_joined.qza'%sr\n",
    "#  table = 'Data/Deblur/%s_arc_table.qza'%sr\n",
    "#  taqzv = 'Data/Deblur/%s_arc_table.qzv'%sr\n",
    "#  repsq = 'Data/Deblur/%s_arc_rep-seqs.qza'%sr\n",
    "#  reqzv = 'Data/Deblur/%s_arc_rep-seqs.qzv'%sr\n",
    "#  stats = 'Data/Deblur/%s_arc_denoising-stats.qza'%sr\n",
    "#  stqzv = 'Data/Deblur/%s_arc_denoising-stats.qzv'%sr\n",
    "  \n",
    "#  !qiime deblur denoise-16S \\\n",
    "#    --i-demultiplexed-seqs $joined \\\n",
    "#    --p-trim-length 300 \\\n",
    "#    --p-jobs-to-start 10 \\\n",
    "#    --o-table $table \\\n",
    "#    --o-representative-sequences $repsq \\\n",
    "#    --o-stats $stats#\n",
    "\n",
    "#  !qiime feature-table tabulate-seqs \\\n",
    "#    --i-data $repsq \\\n",
    "#    --o-visualization $reqzv\n",
    "\n",
    "#  !qiime feature-table summarize \\\n",
    "#    --i-table $table \\\n",
    "#    --o-visualization $taqzv \\\n",
    "#    --m-sample-metadata-file metadata.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Dada2 denoising"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Denoised/SeqRun_2_arc_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: Data/Denoised/SeqRun_2_arc_rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[DADA2Stats] to: Data/Denoised/SeqRun_2_arc_denoising-stats.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Denoised/SeqRun_2_arc_rep-seqs.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Denoised/SeqRun_2_arc_denoising-stats.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Denoised/SeqRun_2_arc_table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "met = pd.read_csv('metadata.tsv', sep='\\t', index_col='#SampleID')\n",
    "\n",
    "adapter = len(met.LinkerPrimerSequence.tolist()[0])\n",
    "!mkdir Data/Denoised\n",
    "for sr in set(met.SeqRun):\n",
    "  cutad = 'Data/Cutadapt/%s_trim.qza'%sr\n",
    "  table = 'Data/Denoised/%s_arc_table.qza'%sr\n",
    "  taqzv = 'Data/Denoised/%s_arc_table.qzv'%sr\n",
    "  repsq = 'Data/Denoised/%s_arc_rep-seqs.qza'%sr\n",
    "  reqzv = 'Data/Denoised/%s_arc_rep-seqs.qzv'%sr\n",
    "  stats = 'Data/Denoised/%s_arc_denoising-stats.qza'%sr\n",
    "  stqzv = 'Data/Denoised/%s_arc_denoising-stats.qzv'%sr\n",
    "  \n",
    "  !qiime dada2 denoise-paired \\\n",
    "    --i-demultiplexed-seqs $cutad \\\n",
    "    --p-trunc-len-f 220 \\\n",
    "    --p-trunc-len-r 240 \\\n",
    "    --p-n-threads 10 \\\n",
    "    --o-table $table \\\n",
    "    --o-representative-sequences $repsq \\\n",
    "    --o-denoising-stats $stats\n",
    "\n",
    "  !qiime feature-table tabulate-seqs \\\n",
    "    --i-data $repsq \\\n",
    "    --o-visualization $reqzv\n",
    "\n",
    "  !qiime metadata tabulate \\\n",
    "    --m-input-file $stats \\\n",
    "    --o-visualization $stqzv\n",
    "\n",
    "  !qiime feature-table summarize \\\n",
    "    --i-table $table \\\n",
    "    --o-visualization $taqzv \\\n",
    "    --m-sample-metadata-file metadata.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Data/Denoised’: File exists\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Denoised/SeqRun_2_arc_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: Data/Denoised/SeqRun_2_arc_rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[DADA2Stats] to: Data/Denoised/SeqRun_2_arc_denoising-stats.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Denoised/SeqRun_2_arc_rep-seqs.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Denoised/SeqRun_2_arc_denoising-stats.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Denoised/SeqRun_2_arc_table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "met = pd.read_csv('metadata.tsv', sep='\\t', index_col='#SampleID')\n",
    "\n",
    "adapter = len(met.LinkerPrimerSequence.tolist()[0])\n",
    "!mkdir Data/Denoised\n",
    "for sr in set(met.SeqRun):\n",
    "  demux = 'Data/Demux/%s_demux.qza'%sr\n",
    "  table = 'Data/Denoised/%s_arc_table.qza'%sr\n",
    "  taqzv = 'Data/Denoised/%s_arc_table.qzv'%sr\n",
    "  repsq = 'Data/Denoised/%s_arc_rep-seqs.qza'%sr\n",
    "  reqzv = 'Data/Denoised/%s_arc_rep-seqs.qzv'%sr\n",
    "  stats = 'Data/Denoised/%s_arc_denoising-stats.qza'%sr\n",
    "  stqzv = 'Data/Denoised/%s_arc_denoising-stats.qzv'%sr\n",
    "  \n",
    "  !qiime dada2 denoise-paired \\\n",
    "    --i-demultiplexed-seqs $demux \\\n",
    "    --p-trim-left-f $adapter \\\n",
    "    --p-trunc-len-f 185 \\\n",
    "    --p-trunc-len-r 230 \\\n",
    "    --p-n-threads 10 \\\n",
    "    --o-table $table \\\n",
    "    --o-representative-sequences $repsq \\\n",
    "    --o-denoising-stats $stats\n",
    "\n",
    "  !qiime feature-table tabulate-seqs \\\n",
    "    --i-data $repsq \\\n",
    "    --o-visualization $reqzv\n",
    "\n",
    "  !qiime metadata tabulate \\\n",
    "    --m-input-file $stats \\\n",
    "    --o-visualization $stqzv\n",
    "\n",
    "  !qiime feature-table summarize \\\n",
    "    --i-table $table \\\n",
    "    --o-visualization $taqzv \\\n",
    "    --m-sample-metadata-file metadata.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge different runs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/merged_table.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#!qiime feature-table merge \\\n",
    "#  --i-tables Data/Denoised/SeqRun_1_bac_table.qza Data/Denoised/SeqRun_2_bac_table.qza \\\n",
    "#  --o-merged-table Data/merged_table.qza\n",
    "\n",
    "#!qiime feature-table merge-seqs \\\n",
    "#  --i-data Data/Denoised/SeqRun_1_bac_rep-seqs.qza Data/Denoised/SeqRun_2_bac_rep-seqs.qza \\\n",
    "#  --o-merged-data Data/merged_rep-seqs.qza\n",
    "\n",
    "#!qiime feature-table summarize \\\n",
    "#  --i-table Data/merged_table.qza \\\n",
    "#  --o-visualization Data/merged_table.qzv \\\n",
    "#  --m-sample-metadata-file metadata.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Training feature classifier \n",
    "### 138 Silva db ref-seqs and taxonomy were obtained from Qiime2 website and produced by rescript"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: \u001b[34mqiime feature-classifier extract-reads\u001b[0m [OPTIONS]\n",
      "\n",
      "  Extract simulated amplicon reads from a reference database. Performs in-\n",
      "  silico PCR to extract simulated amplicons from reference sequences that\n",
      "  match the input primer sequences (within the mismatch threshold specified\n",
      "  by `identity`). Both primer sequences must be in the 5' -> 3' orientation.\n",
      "  Sequences that fail to match both primers will be excluded. Reads are\n",
      "  extracted, trimmed, and filtered in the following order: 1. reads are\n",
      "  extracted in specified orientation; 2. primers are removed; 3. reads\n",
      "  longer than `max_length` are removed; 4. reads are trimmed with\n",
      "  `trim_right`; 5. reads are truncated to `trunc_len`; 6. reads are trimmed\n",
      "  with `trim_left`; 7. reads shorter than `min_length` are removed.\n",
      "\n",
      "\u001b[1mInputs\u001b[0m:\n",
      "  \u001b[34m\u001b[4m--i-sequences\u001b[0m ARTIFACT \u001b[32mFeatureData[Sequence]\u001b[0m\n",
      "                                                                    \u001b[35m[required]\u001b[0m\n",
      "\u001b[1mParameters\u001b[0m:\n",
      "  \u001b[34m\u001b[4m--p-f-primer\u001b[0m TEXT       forward primer sequence (5' -> 3').       \u001b[35m[required]\u001b[0m\n",
      "  \u001b[34m\u001b[4m--p-r-primer\u001b[0m TEXT       reverse primer sequence (5' -> 3'). Do not use\n",
      "                          reverse-complemented primer sequence.     \u001b[35m[required]\u001b[0m\n",
      "  \u001b[34m--p-trim-right\u001b[0m INTEGER  \u001b[4mtrim-right\u001b[0m nucleotides are removed from the 3' end\n",
      "                          if \u001b[4mtrim-right\u001b[0m is positive. Applied before \u001b[4mtrunc-len\u001b[0m\n",
      "                          and \u001b[4mtrim-left\u001b[0m.                          \u001b[35m[default: 0]\u001b[0m\n",
      "  \u001b[34m--p-trunc-len\u001b[0m INTEGER   read is cut to \u001b[4mtrunc-len\u001b[0m if \u001b[4mtrunc-len\u001b[0m is positive.\n",
      "                          Applied after \u001b[4mtrim-right\u001b[0m but before \u001b[4mtrim-left\u001b[0m.\n",
      "                                                                  \u001b[35m[default: 0]\u001b[0m\n",
      "  \u001b[34m--p-trim-left\u001b[0m INTEGER   \u001b[4mtrim-left\u001b[0m nucleotides are removed from the 5' end\n",
      "                          if \u001b[4mtrim-left\u001b[0m is positive. Applied after \u001b[4mtrim-right\u001b[0m\n",
      "                          and \u001b[4mtrunc-len\u001b[0m.                          \u001b[35m[default: 0]\u001b[0m\n",
      "  \u001b[34m--p-identity\u001b[0m NUMBER     minimum combined primer match identity threshold.\n",
      "                                                                \u001b[35m[default: 0.8]\u001b[0m\n",
      "  \u001b[34m--p-min-length\u001b[0m INTEGER  Minimum amplicon length. Shorter amplicons are\n",
      "    \u001b[32mRange(0, None)\u001b[0m        discarded. Applied after trimming and truncation, so\n",
      "                          be aware that trimming may impact sequence\n",
      "                          retention. Set to zero to disable min length\n",
      "                          filtering.                             \u001b[35m[default: 50]\u001b[0m\n",
      "  \u001b[34m--p-max-length\u001b[0m INTEGER  Maximum amplicon length. Longer amplicons are\n",
      "    \u001b[32mRange(0, None)\u001b[0m        discarded. Applied before trimming and truncation,\n",
      "                          so plan accordingly. Set to zero (default) to\n",
      "                          disable max length filtering.           \u001b[35m[default: 0]\u001b[0m\n",
      "  \u001b[34m--p-n-jobs\u001b[0m INTEGER      Number of seperate processes to run.\n",
      "    \u001b[32mRange(1, None)\u001b[0m                                                \u001b[35m[default: 1]\u001b[0m\n",
      "  \u001b[34m--p-batch-size\u001b[0m VALUE \u001b[32mInt % Range(1, None) | Str % Choices('auto')\u001b[0m\n",
      "                          Number of sequences to process in a batch. The\n",
      "                          `auto` option is calculated from the number of\n",
      "                          sequences and number of jobs specified.\n",
      "                                                             \u001b[35m[default: 'auto']\u001b[0m\n",
      "  \u001b[34m--p-read-orientation\u001b[0m TEXT \u001b[32mChoices('both', 'forward', 'reverse')\u001b[0m\n",
      "                          Orientation of primers relative to the sequences:\n",
      "                          \"forward\" searches for primer hits in the forward\n",
      "                          direction, \"reverse\" searches reverse-complement,\n",
      "                          and \"both\" searches both directions.\n",
      "                                                             \u001b[35m[default: 'both']\u001b[0m\n",
      "\u001b[1mOutputs\u001b[0m:\n",
      "  \u001b[34m\u001b[4m--o-reads\u001b[0m ARTIFACT \u001b[32mFeatureData[Sequence]\u001b[0m\n",
      "                                                                    \u001b[35m[required]\u001b[0m\n",
      "\u001b[1mMiscellaneous\u001b[0m:\n",
      "  \u001b[34m--output-dir\u001b[0m PATH       Output unspecified results to a directory\n",
      "  \u001b[34m--verbose\u001b[0m / \u001b[34m--quiet\u001b[0m     Display verbose output to stdout and/or stderr\n",
      "                          during execution of this action. Or silence output\n",
      "                          if execution is successful (silence is golden).\n",
      "  \u001b[34m--examples\u001b[0m              Show usage examples and exit.\n",
      "  \u001b[34m--citations\u001b[0m             Show citations and exit.\n",
      "  \u001b[34m--help\u001b[0m                  Show this message and exit.\n",
      "\n",
      "\u001b[33m                  There were some problems with the command:                  \u001b[0m\n",
      "\u001b[31m\u001b[1m (1/2) Missing option '--p-f-primer'.\u001b[0m\n",
      "\u001b[31m\u001b[1m (2/2) Missing option '--p-r-primer'.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# Extract reference reads\n",
    "#!qiime feature-classifier extract-reads \\\n",
    "#  --i-sequences Classifier/silva-138-99-seqs.qza \\\n",
    "#  --p-f-primer AGAGTTTGATCCTGGCTCAG \\\n",
    "#  --p-r-primer TGCTGCCTCCCGTAGGAGT \\\n",
    "#  --p-min-length 280 \\\n",
    "#  --p-max-length 500 \\\n",
    "#  --o-reads Classifier/ref-seqs.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved TaxonomicClassifier to: Classifier/V1-V2_silva_138_classifier.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# Train classifier\n",
    "!qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "  --i-reference-reads Classifier/silva-138-99-seqs.qza \\\n",
    "  --i-reference-taxonomy Classifier/silva-138-99-tax.qza \\\n",
    "  --o-classifier Classifier/full_silva_138_classifier.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Taxonomy assignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Taxonomy] to: Data/taxonomy_vsearch-sklearn.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/taxonomy_vsearch-sklearn.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-classifier classify-hybrid-vsearch-sklearn \\\n",
    "  --i-query Data/Denoised/SeqRun_2_arc_rep-seqs.qza \\\n",
    "  --i-reference-reads Classifier/silva-138-99-seqs.qza \\\n",
    "  --i-reference-taxonomy Classifier/silva-138-99-tax.qza \\\n",
    "  --i-classifier Classifier/full_silva_138_classifier.qza \\\n",
    "  --p-threads 1 \\\n",
    "  --p-no-prefilter \\\n",
    "  --o-classification Data/taxonomy_vsearch-sklearn.qza\n",
    "\n",
    "!qiime metadata tabulate \\\n",
    "  --m-input-file Data/taxonomy_vsearch-sklearn.qza \\\n",
    "  --o-visualization Data/taxonomy_vsearch-sklearn.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Taxonomy] to: Data/taxonomy_blast.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#!qiime feature-classifier classify-consensus-blast \\\n",
    "#  --i-query Data/merged_rep-seqs.qza \\\n",
    "#  --i-reference-reads Classifier/ref-seqs.qza \\\n",
    "#  --i-reference-taxonomy Classifier/silva-138-99-tax.qza \\\n",
    "#  --o-classification Data/taxonomy_blast.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combo: Combining ASV hashes with last available taxa "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Install biopython\n",
    "!pip install biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Biom’: File exists\n",
      "mkdir: cannot create directory ‘Taxa’: File exists\n",
      "mkdir: cannot create directory ‘Rep-seqs’: File exists\n",
      "\u001b[32mExported Data/Denoised/SeqRun_2_arc_table.qza as BIOMV210DirFmt to directory Biom/\u001b[0m\n",
      "\u001b[32mExported Data/taxonomy_vsearch-sklearn.qza as TSVTaxonomyDirectoryFormat to directory Taxa\u001b[0m\n",
      "\u001b[32mExported Data/Denoised/SeqRun_2_arc_rep-seqs.qza as DNASequencesDirectoryFormat to directory Rep-seqs/\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "# declare Qiime2 generated artifacts here\n",
    "table    = 'Data/Denoised/SeqRun_2_arc_table.qza' \n",
    "taxo     = 'Data/taxonomy_vsearch-sklearn.qza'\n",
    "rep_seq  = 'Data/Denoised/SeqRun_2_arc_rep-seqs.qza'\n",
    "\n",
    "# export rep-seqs.qza, table.qza and taxonomy.qza\n",
    "!mkdir Biom Taxa Rep-seqs #temp directories\n",
    "\n",
    "!qiime tools export \\\n",
    "  --input-path $table \\\n",
    "  --output-path Biom/\n",
    "\n",
    "!qiime tools export \\\n",
    "  --input-path $taxo \\\n",
    "  --output-path Taxa\n",
    "\n",
    "!qiime tools export \\\n",
    "  --input-path $rep_seq \\\n",
    "  --output-path Rep-seqs/\n",
    "\n",
    "# convert .biom to .tsv\n",
    "!biom convert -i Biom/feature-table.biom -o Biom/feature-table.tsv --to-tsv \n",
    "\n",
    "# *****replacing hashes with combination of taxonomy and beginings of the hashes*****\n",
    "# reading tables\n",
    "taxa = pd.read_csv('Taxa/taxonomy.tsv', sep='\\t')\n",
    "biom = pd.read_csv('Biom/feature-table.tsv', sep='\\t', skiprows=1)\n",
    "\n",
    "#creating a new column with modified taxonomy\n",
    "#I also shortened some annotations and deleted some symbols \n",
    "#that were crashing tree construction with modified files\n",
    "taxa['Taxon'] = taxa.Taxon.replace(';__','').str.replace('[','').str.replace(']','').str.replace('.','')\\\n",
    ".str.replace('/','_').str.replace(\"'\",'').str.replace(' ','_').str.replace('archaeon_enrichment','arc_enrich')\\\n",
    ".str.replace('uncultured_rumen','unc_rum').str.replace('uncultured_archaeon','unc_arc').str.replace('_archaeon','_arc')\\\n",
    ".str.replace('uncultured_compost','unc_comp').str.replace('uncultured_euryarchaeote','unc_euryarc')\\\n",
    ".str.replace('uncultured','unc').str.replace('unidentified','unid')\n",
    "\n",
    "#dealing with uncultured taxa to provide additional information\n",
    "taxa['Combo'] =  taxa['Taxon'].str.split(\"__\").str[-1].str.split(\";\").str[-1]\n",
    "for x in ['unc_euryarc','unc_rum','unid_methanogen','arc_enrich','unc_arc','unc_comp','unc']:\n",
    "  taxa.loc[taxa['Combo'].str[:]==x,'Combo']=taxa['Taxon'].str.split(\"__\").str[-2].str.split(';').str[0]+'_'+x\n",
    "  \n",
    "for n in range(3,6):\n",
    "  taxa.loc[taxa.Combo.str[:]=='unc_unc_arc','Combo']=taxa.Taxon.str.split(\"__\").str[-n].str.split(';').str[0]+'_unc_arc'\n",
    "  taxa.loc[taxa.Combo.str[:]=='unc_unc_rum','Combo']=taxa.Taxon.str.split(\"__\").str[-n].str.split(';').str[0]+'_unc_rum'\n",
    "  taxa.loc[taxa.Combo.str[:]=='unc_unc','Combo']=taxa.Taxon.str.split(\"__\").str[-n].str.split(';').str[0]+'_unc'\n",
    "  \n",
    "#add modified taxonomy information to feature hashes, separating them by '|'\n",
    "biom['#OTU ID'] = taxa['Combo']+'|'+taxa['Feature ID']\n",
    "taxa['Feature ID'] = biom['#OTU ID']\n",
    "taxa = taxa[['Feature ID', 'Taxon', 'Confidence', 'Consensus', 'Method']]\n",
    "\n",
    "### writing modified files\n",
    "biom.to_csv('Biom/feature-table.tsv', sep='\\t', index=False)\n",
    "taxa.to_csv('Taxa/taxonomy.tsv', sep='\\t', index=False)\n",
    "fasta_hash  = r\"Rep-seqs/dna-sequences.fasta\"\n",
    "fasta_combo = r\"Rep-seqs/dna-sequences.fa\"\n",
    "hlist = biom['#OTU ID'].tolist()\n",
    "with open(fasta_hash) as hashes, open(fasta_combo, 'w') as combo:\n",
    "  for record in SeqIO.parse(fasta_hash, 'fasta'):\n",
    "    for h in hlist:\n",
    "      if str(record.id) in h:\n",
    "        combo.write('>'+h+'\\n'+str(record.seq)+'\\n')\n",
    "\n",
    "#some cleaning and renaming\n",
    "!rm $fasta_hash\n",
    "!mv $fasta_combo $fasta_hash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mImported Biom/feature-table.biom as BIOMV210Format to Data/combo_table.qza\u001b[0m\n",
      "\u001b[32mImported Taxa/taxonomy.tsv as TSVTaxonomyDirectoryFormat to Data/combo_taxonomy.qza\u001b[0m\n",
      "\u001b[32mImported Rep-seqs/dna-sequences.fasta as DNASequencesDirectoryFormat to Data/combo_rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/combo_table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#creating new rep-seqs.qza, table.qza and taxonomy.qza with modified hashes (added 'combo_' in the name)\n",
    "!biom convert -i Biom/feature-table.tsv -o Biom/feature-table.biom --table-type=\"OTU table\" --to-hdf5\n",
    "\n",
    "!qiime tools import \\\n",
    "  --input-path Biom/feature-table.biom \\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --input-format BIOMV210Format \\\n",
    "  --output-path Data/combo_table.qza\n",
    "\n",
    "!qiime tools import \\\n",
    "  --type 'FeatureData[Taxonomy]' \\\n",
    "  --input-path Taxa/taxonomy.tsv \\\n",
    "  --output-path Data/combo_taxonomy.qza\n",
    "\n",
    "!qiime tools import \\\n",
    "  --input-path $fasta_hash \\\n",
    "  --type 'FeatureData[Sequence]' \\\n",
    "  --output-path Data/combo_rep-seqs.qza\n",
    "\n",
    "!rm -r Biom Taxa Rep-seqs #clean temp directories\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "  --i-table Data/combo_table.qza \\\n",
    "  --m-sample-metadata-file metadata.tsv \\\n",
    "  --o-visualization Data/combo_table.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate a tree for phylogenetic diversity analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[AlignedSequence] to: Data/aligned-rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[AlignedSequence] to: Data/masked-aligned-rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved Phylogeny[Unrooted] to: Data/unrooted-tree.qza\u001b[0m\n",
      "\u001b[32mSaved Phylogeny[Rooted] to: Data/rooted-tree.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "  --i-sequences Data/combo_rep-seqs.qza \\\n",
    "  --p-n-threads 10 \\\n",
    "  --o-alignment Data/aligned-rep-seqs.qza \\\n",
    "  --o-masked-alignment Data/masked-aligned-rep-seqs.qza \\\n",
    "  --o-tree Data/unrooted-tree.qza \\\n",
    "  --o-rooted-tree Data/rooted-tree.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filtration to remove low abundant features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Data/Divided_tables’: File exists\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/full-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/full-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/full-table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Divided_tables/full-table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "tabdir = 'Data/Divided_tables'\n",
    "\n",
    "!mkdir $tabdir\n",
    "!qiime feature-table filter-features \\\n",
    "  --i-table Data/combo_table.qza \\\n",
    "  --p-min-frequency 5 \\\n",
    "  --o-filtered-table $tabdir/full-table.qza\n",
    "\n",
    "!qiime taxa filter-table \\\n",
    "  --i-table $tabdir/full-table.qza \\\n",
    "  --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "  --p-exclude mitochondria,chloroplast,d__Bacteria \\\n",
    "  --p-include p__ \\\n",
    "  --o-filtered-table $tabdir/full-table.qza\n",
    "\n",
    "!qiime feature-table filter-samples \\\n",
    "  --i-table $tabdir/full-table.qza \\\n",
    "  --p-min-features 5 \\\n",
    "  --p-min-frequency 10 \\\n",
    "  --o-filtered-table $tabdir/full-table.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "  --i-table $tabdir/full-table.qza \\\n",
    "  --o-visualization $tabdir/full-table.qzv \\\n",
    "  --m-sample-metadata-file metadata.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Separating table by day and body site for the analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def filt_smpl(tabin,col,tabout):\n",
    "  !qiime feature-table filter-samples \\\n",
    "    --i-table $tabin \\\n",
    "    --m-metadata-file metadata.tsv \\\n",
    "    --p-where \"$col\" \\\n",
    "    --o-filtered-table $tabout\n",
    "\n",
    "def filt_feat(tabin,minfrq,tabout):\n",
    "  !qiime feature-table filter-features \\\n",
    "    --i-table $tabin \\\n",
    "    --p-min-frequency $minfrq \\\n",
    "    --o-filtered-table $tabout\n",
    "    \n",
    "def tabqzv(tabin,tabout):\n",
    "  !qiime feature-table summarize \\\n",
    "    --i-table $tabin \\\n",
    "    --o-visualization $tabout \\\n",
    "    --m-sample-metadata-file metadata.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/d0-RSP_RF-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/d0-RSP_RF-table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Divided_tables/d0-RSP_RF-table.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/d7-E_FL-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/d7-E_FL-table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Divided_tables/d7-E_FL-table.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/d13-E_FL_SAM_FR-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/d13-E_FL_SAM_FR-table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Divided_tables/d13-E_FL_SAM_FR-table.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/d7_d13-E_FL-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Divided_tables/d7_d13-E_FL-table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Divided_tables/d7_d13-E_FL-table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# divide by days \n",
    "tabdir = 'Data/Divided_tables'\n",
    "\n",
    "for t in ['d0-RSP_RF','d7-E_FL','d13-E_FL_SAM_FR', 'd7_d13-E_FL']:\n",
    "  day = \"[Day]='%s'\" % t.split('-')[0]\n",
    "  tab = tabdir+'/%s-table.qza' % t\n",
    "  tav = tabdir+'/%s-table.qzv' % t\n",
    "  if t == 'd7_d13-E_FL':\n",
    "    day = \"[Day] IN ('d7','d13') AND NOT [BS] IN ('SAM','FR')\"\n",
    "  filt_smpl(tabdir+'/full-table.qza',day,tab)\n",
    "  filt_feat(tab,5,tab)\n",
    "  tabqzv(tab,tav)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### List of all tables produced"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "tabdir = 'Data/Divided_tables'\n",
    "tables = ['d0-RSP_RF','d7-E_FL','d13-E_FL_SAM_FR','d7_d13-E_FL','full']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Taxabarplots"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### By samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_samples/d0-RSP_RF-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_samples/d7-E_FL-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_samples/d13-E_FL_SAM_FR-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_samples/d7_d13-E_FL-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_samples/full-taxabarplot.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!mkdir -p Results/Taxa_barplots/By_samples\n",
    "tabdir = 'Data/Divided_tables'\n",
    "tables = ['d0-RSP_RF','d7-E_FL','d13-E_FL_SAM_FR','d7_d13-E_FL','full']\n",
    "for tab in tables:\n",
    "  table = tabdir+'/%s-table.qza' % tab\n",
    "  out = 'Results/Taxa_barplots/By_samples/%s-taxabarplot.qzv' % tab\n",
    "  \n",
    "  !qiime taxa barplot \\\n",
    "    --i-table $table \\\n",
    "    --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "    --m-metadata-file metadata.tsv \\\n",
    "    --o-visualization $out"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### By groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-Day_hour-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-Day_hour-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-Treatment-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-Treatment-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-Source-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-Source-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Dh-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Dh-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Day_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Day_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Day_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Day_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Dh_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Dh_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Dh_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Dh_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Day_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Day_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d0-RSP_RF-BS_Dh_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d0-RSP_RF-BS_Dh_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-Day_hour-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-Day_hour-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-Treatment-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-Treatment-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-Source-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-Source-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Dh-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Dh-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Day_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Day_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Day_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Day_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Dh_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Dh_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Dh_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Dh_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Day_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Day_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7-E_FL-BS_Dh_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7-E_FL-BS_Dh_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-Day_hour-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-Day_hour-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-Treatment-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-Treatment-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-Source-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-Source-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Dh-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Dh-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Day_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Day_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Day_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Day_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Dh_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Dh_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Dh_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Dh_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Day_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Day_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d13-E_FL_SAM_FR-BS_Dh_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d13-E_FL_SAM_FR-BS_Dh_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-Day_hour-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-Day_hour-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-Treatment-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-Treatment-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-Source-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-Source-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Dh-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Dh-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Day_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Day_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Day_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Day_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Dh_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Dh_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Dh_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Dh_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Day_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Day_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/d7_d13-E_FL-BS_Dh_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/d7_d13-E_FL-BS_Dh_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-Day_hour-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-Day_hour-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-Treatment-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-Treatment-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-Source-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-Source-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Day-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Day-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Dh-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Dh-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Day_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Day_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Day_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Day_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Dh_Trt-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Dh_Trt-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Dh_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Dh_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Day_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Day_Trt_Src-taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Grouped_tables/full-BS_Dh_Trt_Src-group_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/By_groups/full-BS_Dh_Trt_Src-taxabarplot.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "tables = ['d0-RSP_RF','d7-E_FL','d13-E_FL_SAM_FR','d7_d13-E_FL','full']\n",
    "tabdir = 'Data/Divided_tables'\n",
    "outres = 'Results/Taxa_barplots/By_groups'\n",
    "outdat = 'Data/Grouped_tables'\n",
    "met = pd.read_csv('metadata.tsv',sep='\\t',index_col=0)\n",
    "trash = ['BarcodeSequence','LinkerPrimerSequence','Run_index','Sample Name','Smpl_no','BodySite',\\\n",
    "         'Date of Sampling','Description','Description 2','SeqRun','Src_rstcRun','Day_num','rstc_run']\n",
    "groups = [col for col in met.columns.tolist() if col not in trash]\n",
    "\n",
    "for tab in tables:\n",
    "  tabin = tabdir+'/%s-table.qza' % tab\n",
    "  \n",
    "  !mkdir -p $outres $outdat\n",
    "  for group in groups:\n",
    "  #Collapse metadata\n",
    "    meta = met.drop_duplicates(subset=group).copy()\n",
    "    meta.index = meta[group]\n",
    "    meta.index.name = '#SampleID'\n",
    "    meta.to_csv(outdat+'/%s-%s-metadata.tsv'%(tab,group),sep='\\t')\n",
    "  #Variables    \n",
    "    grouped = outdat + '/%s-%s-group_table.qza' % (tab,group)\n",
    "    taxabar = outres + '/%s-%s-taxabarplot.qzv' % (tab,group)\n",
    "    metadata= outdat + '/%s-%s-metadata.tsv'    % (tab,group)\n",
    "    \n",
    "    try:\n",
    "    #Group tables    \n",
    "      !qiime feature-table group \\\n",
    "        --i-table $tabin \\\n",
    "        --p-axis 'sample' \\\n",
    "        --m-metadata-file metadata.tsv \\\n",
    "        --m-metadata-column $group \\\n",
    "        --p-mode 'mean-ceiling' \\\n",
    "        --o-grouped-table $grouped\n",
    "    #Taxabarplots    \n",
    "      !qiime taxa barplot \\\n",
    "        --i-table $grouped \\\n",
    "        --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "        --m-metadata-file $metadata \\\n",
    "        --o-visualization $taxabar\n",
    "    except:\n",
    "      print('Looks like there is no %s group in table %s' % (group, tabin))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alpha and beta diversity analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Alpha rarefaction plotting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Results/Alpha_rarefaction.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime diversity alpha-rarefaction \\\n",
    "  --i-table Data/Divided_tables/full-table.qza \\\n",
    "  --i-phylogeny Data/rooted-tree.qza \\\n",
    "  --p-max-depth 30000 \\\n",
    "  --m-metadata-file metadata.tsv \\\n",
    "  --o-visualization Results/Alpha_rarefaction.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Core-metrics-phylogenetic: Core diversity metrics (phylogenetic and non-phylogenetic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/Core-metrics/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics/faith_pd_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics/observed_features_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Results/Core-metrics/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Results/Core-metrics/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Results/Core-metrics/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Results/Core-metrics/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics/unweighted_unifrac_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics/weighted_unifrac_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics/jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics/bray_curtis_emperor.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime diversity core-metrics-phylogenetic \\\n",
    "  --i-phylogeny Data/rooted-tree.qza \\\n",
    "  --i-table Data/Divided_tables/full-table.qza \\\n",
    "  --p-sampling-depth 4455 \\\n",
    "  --m-metadata-file metadata.tsv \\\n",
    "  --p-n-jobs-or-threads 'auto' \\\n",
    "  --output-dir Results/Core-metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Principal Coordinate Analysis Biplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Data/Relative_tables’: File exists\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Relative_tables/full-relative_table.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Results/Biplots/Biplots_full/unweighted_unifrac_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Biplots/Biplots_full/unweighted_unifrac_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Results/Biplots/Biplots_full/weighted_unifrac_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Biplots/Biplots_full/weighted_unifrac_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Results/Biplots/Biplots_full/jaccard_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Biplots/Biplots_full/jaccard_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Results/Biplots/Biplots_full/bray_curtis_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Biplots/Biplots_full/bray_curtis_biplot.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "table = 'Data/Divided_tables/full-table.qza'\n",
    "reltab = 'Data/Relative_tables/full-relative_table.qza'\n",
    "  \n",
    "  # Converting feature table [Frequency] to [Relative frequency]\n",
    "!mkdir Data/Relative_tables\n",
    "!qiime feature-table relative-frequency \\\n",
    "  --i-table Data/Divided_tables/full-table.qza \\\n",
    "  --o-relative-frequency-table $reltab\n",
    "\n",
    "for artifact in ['unweighted_unifrac', 'weighted_unifrac', 'jaccard', 'bray_curtis']:\n",
    "  pcoa  = 'Results/Core-metrics/%s_pcoa_results.qza' % artifact\n",
    "  bidir = 'Results/Biplots/Biplots_full'\n",
    "  biplot= bidir+'/%s_biplot.qza' % artifact\n",
    "  bi_qzv= bidir+'/%s_biplot.qzv' % artifact\n",
    "\n",
    "  !mkdir -p $bidir\n",
    "    \n",
    "  # pcoa-biplot: Principal Coordinate Analysis Biplot     \n",
    "  !qiime diversity pcoa-biplot \\\n",
    "    --i-pcoa $pcoa \\\n",
    "    --i-features $reltab \\\n",
    "    --o-biplot $biplot\n",
    "\n",
    "  !qiime emperor biplot \\\n",
    "    --i-biplot $biplot \\\n",
    "    --m-sample-metadata-file metadata.tsv \\\n",
    "    --p-ignore-missing-samples \\\n",
    "    --p-number-of-features 5 \\\n",
    "    --o-visualization $bi_qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Filtering beta diversity distances"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Beta diversity comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BodySite_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/rstc_run_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-vh0e01dn.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-bj699yz6.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-6vgmk4ir.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/Source_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Trt_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Trt_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Trt_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BodySite_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/rstc_run_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-dsbqoqpv.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-pnag9olv.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-18fq0v0a.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/Source_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Trt_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Trt_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Trt_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BodySite_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/rstc_run_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-948f40w9.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-nr2lukj5.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-2atemmsw.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/Source_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Trt_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Trt_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Trt_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BodySite_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/rstc_run_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-sqp8053z.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-d4n__xij.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-h78v6db4.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/Source_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Trt_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Day_Trt_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d0-RSP_RF/BS_Dh_Trt_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BodySite_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/rstc_run_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-o5yph2gc.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-ro8w5oix.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/Treatment_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/Source_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Trt_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Trt_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Trt_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BodySite_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/rstc_run_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-flpqt56n.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-51rl6huq.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/Treatment_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/Source_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Trt_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Trt_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Trt_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BodySite_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/rstc_run_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-t2v6c2sn.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-smj6aovm.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/Treatment_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/Source_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Trt_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Trt_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Trt_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BodySite_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/rstc_run_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-5e4dopib.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-1awa2j8_.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/Treatment_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/Source_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Trt_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Day_Trt_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7-E_FL/BS_Dh_Trt_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BodySite_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/rstc_run_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-9dnt7gdg.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Day_hour_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Treatment_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Source_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-emiqmj9j.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-z4zjisgc.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-xg7m0cee.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BodySite_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/rstc_run_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-l9kuzwto.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Day_hour_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Treatment_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Source_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-ppuuuhki.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-u31r2dla.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-s4mciho8.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BodySite_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/rstc_run_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-k3eev55v.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Day_hour_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Treatment_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Source_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-hsm4fuij.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-3yh6rbcq.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-y36ybsht.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BodySite_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/rstc_run_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-35zqx4r1.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Day_hour_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Treatment_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/Source_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Day_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d13-E_FL_SAM_FR/BS_Dh_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-n_svjufr.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-hwc3qkjb.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-16rrvvsr.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BodySite_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/rstc_run_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Day_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-fy1ahjpy.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Treatment_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Source_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Trt_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-13i19e02.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Trt_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BodySite_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/rstc_run_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Day_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-5scyz9lh.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Treatment_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Source_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Trt_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-uaxnpv6z.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Trt_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BodySite_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/rstc_run_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Day_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-rmiou51m.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Treatment_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Source_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Trt_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-0gub5ih2.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Trt_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BodySite_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/rstc_run_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Day_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are the same. This method cannot operate on a grouping vector with only a single group of objects (e.g., there are no 'between' distances because there is only a single group).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-bu6n7gfh.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Treatment_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/Source_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Day_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Trt_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-1ijg78cl.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_d7_d13-E_FL/BS_Dh_Trt_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BodySite_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/rstc_run_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Day_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Day_hour_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Treatment_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Source_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_Trt_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_Src_unweighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-lrpkmmcd.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-79fg_gmg.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-hfkj4o5n.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BodySite_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/rstc_run_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Day_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Day_hour_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Treatment_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Source_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_Trt_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_Src_weighted_unifrac_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-l0h802xy.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-p_xhkwb_.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-pf3zmgbk.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BodySite_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/rstc_run_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Day_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Day_hour_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Treatment_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Source_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_Trt_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_Src_jaccard_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-yef3cggr.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-wdmny5d5.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-7dql9isl.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BodySite_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/rstc_run_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Day_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Day_hour_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Treatment_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/Source_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Day_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_Trt_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Beta_comparisons/Beta_compar_full/BS_Dh_Src_bray_curtis_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-6oqxhdj4.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-_2w0hdqp.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  All values in the grouping vector are unique. This method cannot operate on a grouping vector with only unique values (e.g., there are no 'within' distances because each group of objects contains only a single object).\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-pi2etx2r.log\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "tabdir = 'Data/Divided_tables'\n",
    "tables = ['d0-RSP_RF','d7-E_FL','d13-E_FL_SAM_FR','d7_d13-E_FL','full']\n",
    "\n",
    "meta = pd.read_csv('metadata.tsv',sep='\\t',index_col=0)\n",
    "trash = ['BarcodeSequence','LinkerPrimerSequence','Run_index','Sample Name','Smpl_no',\\\n",
    "         'Date of Sampling','Description','Description 2','SeqRun','Src_rstcRun','Day_num']\n",
    "groups = [col for col in meta.columns.tolist() if col not in trash]\n",
    "\n",
    "for tab in tables:\n",
    "  for beta in ['unweighted_unifrac', 'weighted_unifrac', 'jaccard', 'bray_curtis']:\n",
    "    outdir  = 'Results/Beta_comparisons/Beta_compar_%s'% tab\n",
    "    distance= 'Results/Core-metrics/Core_%s/%s_distance_matrix.qza' % (tab,beta)\n",
    "    \n",
    "    !mkdir -p $outdir\n",
    "    \n",
    "    for column in groups:\n",
    "      bgs_qzv = outdir+'/%s_%s_beta-group-sign-pairwise.qzv' % (column, beta)\n",
    "   \n",
    "      !qiime diversity beta-group-significance \\\n",
    "        --i-distance-matrix $distance \\\n",
    "        --m-metadata-file metadata.tsv \\\n",
    "        --m-metadata-column $column \\\n",
    "        --p-pairwise \\\n",
    "        --o-visualization $bgs_qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Adonis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  Command '['run_adonis.R', '/tmp/tmpz27137f5/dm.tsv', '/tmp/tmpz27137f5/md.tsv', 'BS*Treatment*Source', '999', '10', '/tmp/qiime2-temp-m8ud9d0h/adonis.tsv']' returned non-zero exit status 1.\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-syli5gkw.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  Command '['run_adonis.R', '/tmp/tmps0bn9jit/dm.tsv', '/tmp/tmps0bn9jit/md.tsv', 'BS*Treatment*Source', '999', '10', '/tmp/qiime2-temp-51a29kho/adonis.tsv']' returned non-zero exit status 1.\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-uuxc8ub4.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  Command '['run_adonis.R', '/tmp/tmpc1dvfqo0/dm.tsv', '/tmp/tmpc1dvfqo0/md.tsv', 'BS*Treatment*Source', '999', '10', '/tmp/qiime2-temp-klu7gjh4/adonis.tsv']' returned non-zero exit status 1.\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-wx1my7nw.log\u001b[0m\n",
      "\u001b[31m\u001b[1mPlugin error from diversity:\n",
      "\n",
      "  Command '['run_adonis.R', '/tmp/tmpqhp36x3o/dm.tsv', '/tmp/tmpqhp36x3o/md.tsv', 'BS*Treatment*Source', '999', '10', '/tmp/qiime2-temp-f50zkk5q/adonis.tsv']' returned non-zero exit status 1.\n",
      "\n",
      "Debug info has been saved to /tmp/qiime2-q2cli-err-il_o5wm4.log\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d7-E_FL/unweighted_unifrac_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d7-E_FL/weighted_unifrac_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d7-E_FL/jaccard_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d7-E_FL/bray_curtis_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d13-E_FL_SAM_FR/unweighted_unifrac_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d13-E_FL_SAM_FR/weighted_unifrac_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d13-E_FL_SAM_FR/jaccard_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d13-E_FL_SAM_FR/bray_curtis_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d7_d13-E_FL/unweighted_unifrac_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d7_d13-E_FL/weighted_unifrac_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d7_d13-E_FL/jaccard_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_d7_d13-E_FL/bray_curtis_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_full/unweighted_unifrac_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_full/weighted_unifrac_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_full/jaccard_adonis.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Adonis/Adonis_full/bray_curtis_adonis.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "tabdir = 'Data/Divided_tables'\n",
    "tables = ['d7-E_FL','d13-E_FL_SAM_FR','d7_d13-E_FL','full']\n",
    "\n",
    "meta = pd.read_csv('metadata.tsv',sep='\\t',index_col=0)\n",
    "trash = ['BarcodeSequence','LinkerPrimerSequence','Run_index','Sample Name','Smpl_no',\\\n",
    "         'Date of Sampling','Description','Description 2','SeqRun','Src_rstcRun','Day_num']\n",
    "groups = [col for col in meta.columns.tolist() if col not in trash]\n",
    "formula = 'BS*Treatment*Source'\n",
    "\n",
    "for tab in tables:\n",
    "  if tab == 'd13-E_FL_SAM_FR':\n",
    "    formula = 'BS*Treatment*Source*Day_hour'\n",
    "  if tab == 'd7_d13-E_FL':\n",
    "    formula = 'BS*Treatment*Source*Day'\n",
    "  for beta in ['unweighted_unifrac', 'weighted_unifrac', 'jaccard', 'bray_curtis']:\n",
    "    outdir  = 'Results/Adonis/Adonis_%s'% tab\n",
    "    distance= 'Results/Core-metrics/Core_%s/%s_distance_matrix.qza' % (tab,beta)\n",
    "    \n",
    "    !mkdir -p $outdir\n",
    "    adonis = outdir+'/%s_adonis.qzv' % beta\n",
    "\n",
    "    !qiime diversity adonis \\\n",
    "      --i-distance-matrix $distance \\\n",
    "      --m-metadata-file metadata.tsv \\\n",
    "      --p-formula \"$formula\" \\\n",
    "      --p-n-jobs 10 \\\n",
    "      --o-visualization $adonis  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ANCOM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Results/ANCOM’: File exists\n"
     ]
    }
   ],
   "source": [
    "!mkdir Results/ANCOM\n",
    "def Ancom(column, ancom):\n",
    "  ctable = 'Results/ANCOM/ctable.qza'\n",
    "  !qiime composition add-pseudocount \\\n",
    "    --i-table Results/ANCOM/table.qza \\\n",
    "    --o-composition-table $ctable\n",
    "  !qiime composition ancom \\\n",
    "    --i-table $ctable \\\n",
    "    --m-metadata-file metadata.tsv \\\n",
    "    --m-metadata-column $column \\\n",
    "    --o-visualization $ancom\n",
    "  !rm $ctable\n",
    "    \n",
    "def Ancol(lev, column, ancol):\n",
    "  ctable = 'Results/ANCOM/ctable.qza'\n",
    "  ltable = 'Results/ANCOM/ltable.qza'\n",
    "  !qiime taxa collapse \\\n",
    "    --i-table Results/ANCOM/table.qza \\\n",
    "    --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "    --p-level $lev \\\n",
    "    --o-collapsed-table $ltable\n",
    "  !qiime composition add-pseudocount \\\n",
    "    --i-table $ltable \\\n",
    "    --o-composition-table $ctable\n",
    "  !qiime composition ancom \\\n",
    "    --i-table $ctable \\\n",
    "    --m-metadata-file metadata.tsv \\\n",
    "    --m-metadata-column $column \\\n",
    "    --o-visualization $ancol\n",
    "  !rm $ctable $ltable \n",
    "  \n",
    "def filter_table(df):\n",
    "  df.index.names = ['SampleID']\n",
    "  df.to_csv('Results/ANCOM/table.tsv',sep='\\t')\n",
    "  !qiime feature-table filter-samples \\\n",
    "    --i-table Data/Divided_tables/full-table.qza \\\n",
    "    --m-metadata-file Results/ANCOM/table.tsv \\\n",
    "    --o-filtered-table Results/ANCOM/table.qza\n",
    "  !rm Results/ANCOM/table.tsv\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_E_BS_Trt_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_E_BS_Trt_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_E_BS_Trt_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_E_BS_Src_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_E_BS_Src_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_E_BS_Src_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_FL_BS_Trt_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_FL_BS_Trt_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_FL_BS_Trt_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_FL_BS_Src_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_FL_BS_Src_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d7_FL_BS_Src_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_E_BS_Trt_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_E_BS_Trt_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_E_BS_Trt_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_E_BS_Src_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_E_BS_Src_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_E_BS_Src_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FL_BS_Trt_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FL_BS_Trt_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FL_BS_Trt_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FL_BS_Src_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FL_BS_Src_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FL_BS_Src_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_SAM_BS_Trt_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_SAM_BS_Trt_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_SAM_BS_Trt_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_SAM_BS_Src_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_SAM_BS_Src_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_SAM_BS_Src_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Trt_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Trt_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Trt_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Src_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Src_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Src_ancom_7.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Dh_ancom.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Dh_ancom_6.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/ANCOM/ltable.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Composition] to: Results/ANCOM/ctable.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/ANCOM/d13_FR_BS_Dh_ancom_7.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from os import path\n",
    "\n",
    "met = pd.read_csv('metadata.tsv',sep='\\t',index_col=0)\n",
    "\n",
    "cols = ['BS_Trt','BS_Src','BS_Dh']\n",
    "days = ['d7','d13']\n",
    "bodysites = ['E','FL','SAM','FR']\n",
    "for day in days:\n",
    "  for bs in bodysites:\n",
    "    if day=='d7' and bs in ['SAM','FR']:\n",
    "      continue\n",
    "    dftab = met.loc[(met.Day==day)&(met.BS==bs)].copy()\n",
    "    filter_table(dftab)\n",
    "    for col in cols:\n",
    "      if col=='BS_Dh' and bs!='FR':\n",
    "        continue\n",
    "        \n",
    "      ancom = 'Results/ANCOM/%s_%s_%s_ancom.qzv'%(day,bs,col)\n",
    "      Ancom(col,ancom)\n",
    "      for lev in [6,7]:\n",
    "        ancol = 'Results/ANCOM/%s_%s_%s_ancom_%s.qzv'%(day,bs,col,lev)\n",
    "        Ancol(lev,col,ancol)\n",
    "    !rm Results/ANCOM/table.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Longitudinal analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pairwise-differences and distances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "def df_matrix(beta,met):\n",
    "  matr = beta[beta.index.isin(met.index)].copy()\n",
    "  matr = matr[met.index]\n",
    "  matr.sort_index(inplace=True)\n",
    "  matr = matr.reindex(sorted(matr.columns),axis=1)\n",
    "  return matr\n",
    "\n",
    "def first_diff(metric,col,diff):\n",
    "  !qiime longitudinal pairwise-differences \\\n",
    "    --m-metadata-file Results/First_differences/alpha.tsv \\\n",
    "    --p-metric $metric \\\n",
    "    --p-group-column $col \\\n",
    "    --p-state-column Day \\\n",
    "    --p-state-1 d7 \\\n",
    "    --p-state-2 d13 \\\n",
    "    --p-individual-id-column IID \\\n",
    "    --p-replicate-handling random \\\n",
    "    --o-visualization $diff\n",
    "  \n",
    "def first_dist(dista,col,distv):\n",
    "  !qiime longitudinal pairwise-distances \\\n",
    "    --i-distance-matrix $dista \\\n",
    "    --m-metadata-file Results/First_differences/alpha.tsv \\\n",
    "    --p-group-column $col \\\n",
    "    --p-state-column Day \\\n",
    "    --p-state-1 d7 \\\n",
    "    --p-state-2 d13 \\\n",
    "    --p-individual-id-column IID \\\n",
    "    --p-replicate-handling random \\\n",
    "    --o-visualization $distv\n",
    "  \n",
    "def distance_matrix(matr,div,col,dista):\n",
    "  matr.to_csv('Results/First_distances/matrix.tsv',sep='\\t')\n",
    "  matr = 'Results/First_distances/matrix.tsv'\n",
    "      \n",
    "  !qiime tools import \\\n",
    "    --input-path $matr \\\n",
    "    --output-path $dista \\\n",
    "    --type DistanceMatrix\n",
    "  !rm $matr\n",
    "\n",
    "def beta_div(qza):  \n",
    "  a = !unzip $qza\n",
    "  digest = a[1].split('/')[0].replace('  inflating: ','')\n",
    "  inf = digest + '/data/distance-matrix.tsv'\n",
    "  data = pd.read_csv(inf, sep='\\t',index_col=0)\n",
    "  !rm -r $digest\n",
    "  return data "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Results/First_differences’: File exists\n",
      "\u001b[32mSaved Visualization to: Results/First_differences/observed_features_BS_Trt_first-diff.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_differences/observed_features_BS_Src_first-diff.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_differences/shannon_entropy_BS_Trt_first-diff.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_differences/shannon_entropy_BS_Src_first-diff.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_differences/faith_pd_BS_Trt_first-diff.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_differences/faith_pd_BS_Src_first-diff.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_differences/pielou_evenness_BS_Trt_first-diff.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_differences/pielou_evenness_BS_Src_first-diff.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "diversity = {'observed_features':'Observed ASVs',\n",
    "             'shannon_entropy':'Shannon\\'s entropy',\n",
    "             'faith_pd':'Faith\\'s PD',\n",
    "             'pielou_evenness':'Pielou\\'s evenness'}\n",
    "cols = ['BS_Trt','BS_Src']\n",
    "\n",
    "df = pd.read_csv('Results/Core-metrics/alpha.tsv',sep='\\t',index_col='#SampleID')\n",
    "alpha = df.loc[(df.Day!='d0')&(df.BS!='SAM')&(df.BS!='FR')].copy()\n",
    "alpha['IID'] = ''\n",
    "for group in set(alpha.BS_Day_Trt_Src):\n",
    "  temp = alpha.loc[alpha.BS_Day_Trt_Src==group].copy()\n",
    "  for i,index in enumerate(temp.index):\n",
    "    alpha.loc[index,'IID'] = temp.loc[index,'BS_Trt_Src']+'_'+str(i)\n",
    "      \n",
    "!mkdir Results/First_differences\n",
    "alpha.to_csv('Results/First_differences/alpha.tsv',sep='\\t')\n",
    "for div in diversity:\n",
    "  for col in cols:\n",
    "    diff = 'Results/First_differences/%s_%s_first-diff.qzv'%(div,col)\n",
    "    first_diff(div,col,diff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Results/First_distances’: File exists\n",
      "\u001b[32mImported Results/First_distances/matrix.tsv as DistanceMatrixDirectoryFormat to Results/First_distances/unweighted_unifrac_BS_Trt_distances.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_distances/unweighted_unifrac_BS_Trt_first-dist.qzv\u001b[0m\n",
      "\u001b[32mImported Results/First_distances/matrix.tsv as DistanceMatrixDirectoryFormat to Results/First_distances/unweighted_unifrac_BS_Src_distances.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_distances/unweighted_unifrac_BS_Src_first-dist.qzv\u001b[0m\n",
      "\u001b[32mImported Results/First_distances/matrix.tsv as DistanceMatrixDirectoryFormat to Results/First_distances/weighted_unifrac_BS_Trt_distances.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_distances/weighted_unifrac_BS_Trt_first-dist.qzv\u001b[0m\n",
      "\u001b[32mImported Results/First_distances/matrix.tsv as DistanceMatrixDirectoryFormat to Results/First_distances/weighted_unifrac_BS_Src_distances.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_distances/weighted_unifrac_BS_Src_first-dist.qzv\u001b[0m\n",
      "\u001b[32mImported Results/First_distances/matrix.tsv as DistanceMatrixDirectoryFormat to Results/First_distances/jaccard_BS_Trt_distances.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_distances/jaccard_BS_Trt_first-dist.qzv\u001b[0m\n",
      "\u001b[32mImported Results/First_distances/matrix.tsv as DistanceMatrixDirectoryFormat to Results/First_distances/jaccard_BS_Src_distances.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_distances/jaccard_BS_Src_first-dist.qzv\u001b[0m\n",
      "\u001b[32mImported Results/First_distances/matrix.tsv as DistanceMatrixDirectoryFormat to Results/First_distances/bray_curtis_BS_Trt_distances.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_distances/bray_curtis_BS_Trt_first-dist.qzv\u001b[0m\n",
      "\u001b[32mImported Results/First_distances/matrix.tsv as DistanceMatrixDirectoryFormat to Results/First_distances/bray_curtis_BS_Src_distances.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/First_distances/bray_curtis_BS_Src_first-dist.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "diversity = {'unweighted_unifrac':'Unweighted UniFrac',\n",
    "             'weighted_unifrac':'Weighted UniFrac',\n",
    "             'jaccard':'Jaccard',\n",
    "             'bray_curtis':'Bray-Curtis'}\n",
    "cols = ['BS_Trt','BS_Src']\n",
    "df = pd.read_csv('Results/Core-metrics/alpha.tsv',sep='\\t',index_col='#SampleID')\n",
    "met = df.loc[(df.Day!='d0')&(df.BS!='SAM')&(df.BS!='FR')].copy()\n",
    "!mkdir Results/First_distances\n",
    "for div in diversity:\n",
    "  beta = beta_div('Results/Core-metrics/%s_distance_matrix.qza'%div)\n",
    "  for col in cols:\n",
    "    distv = 'Results/First_distances/%s_%s_first-dist.qzv'%(div,col)\n",
    "    dista = 'Results/First_distances/%s_%s_distances.qza'%(div,col)\n",
    "    matr = df_matrix(beta,met)\n",
    "    distance_matrix(matr,div,col,dista)\n",
    "    first_dist(dista,col,distv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
